tests/Propagation_TEST.cxx

```diff
@@ -12,7 +12,7 @@ TEST(Comparison, Comparison_equal)
 {
     Sector::Definition sector_def;
     sector_def.location = Sector::ParticleLocation::InsideDetector;
-    sector_def.SetMedium(std::make_shared<Medium>(Water()));
+    auto medium = std::make_shared<const Water>();
```

Comment:
shouldn't this line be
`sector_def.SetMedium(std::make_shared<const Water>());`